xlators/mgmt/glusterd/src/glusterd-store.c

```diff
@@ -4339,20 +4339,24 @@ glusterd_store_create_peer_dir()
     return ret;
 }
 
-static void
+static int
 glusterd_store_uuid_peerpath_set(glusterd_peerinfo_t *peerinfo, char *peerfpath,
                                  size_t len)
 {
     char peerdir[PATH_MAX];
     char str[50] = {0};
+    int ret = -1;
 
     GF_ASSERT(peerinfo);
     GF_ASSERT(peerfpath);
     GF_ASSERT(len >= PATH_MAX);
 
     glusterd_store_peerinfo_dirpath_set(peerdir, sizeof(peerdir));
     gf_uuid_unparse(peerinfo->uuid, str);
-    snprintf(peerfpath, len, "%s/%s", peerdir, str);
+    ret = snprintf(peerfpath, len, "%s/%s", peerdir, str);
+    if (ret < 0 || ret >= len)
+        return -1;
+    return 0;
```

Comment:
Even better:
```
if (ret)
    return -1;
return 0;
```

Reply:
@nik-redhat  wouldn't using if ( ret ) would return -1 even for the value returned by snprint which are within the buffer range which would not be correct?

Reply:
Ahh....right.
I was in the assumption, that it returns only success and failure values.
Ignore this comment.